File: log_sniffer/src/models/Log.ts

export enum OS {
    WINDOWS = 'windows',
    LINUX = 'linux'
}


export enum Type {
    SYSTEM = 'system',
    APP = 'application',
    LOGGER = 'logger'
}


export default interface Log {
    id: string
    type: Type
    os: OS
    message: string
    hostname: string
    unit: string
    raw: string
    severity: number
    timestamp: number
}

File: cloudlog_commons/cloudlog_commons/shared/log.py
from decimal import Decimal
from dataclasses import dataclass
from enum import Enum


class OS(str, Enum):
    WINDOWS = "windows"
    LINUX = "linux"


class LogType(str, Enum):
    SYSTEM = "system"
    APP = "application"
    LOGGER = "logger"


@dataclass
class Log:
    os: OS
    severity: int
    message: str
    timestamp: Decimal
    hostname: str
    unit: str
    raw: str
    type: LogType
